### PR TITLE
Fix publish diff invalid

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
修复资源diff bug

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by zhu327 — APPROVED** [review]

**Comment by zhu327** [comment]
有空可以跟踪下数据的来源

**Review by wklken — APPROVED** [review]

File: pkg/apisix/synchronizer/etcd/diff_test.go [hunk 1]
```diff
@@ -331,6 +331,69 @@ var _ = Describe("configDiffer", func() {
...
+
+		Context("Test diff Routes", func() {
+			It("diff Routes", func() {
+				put, del := differ.diffRoutes(newRoutes, oldRoutes)
+				Expect(len(put)).To(Equal(0))
+				Expect(len(del)).To(Equal(0))
+			})
+		})
+	})
+
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply pkg/apisix/synchronizer/etcd/diff_test.go:396]: 新增这种不确定类型的map[string]interface{} 这种是可以统一消除差异的。diff没有问题。done

File: pkg/apisix/synchronizer/synchronizer.go [hunk 2]
```diff
@@ -159,7 +159,7 @@ func (as *ApisixConfigurationSynchronizer) flush(ctx context.Context) {
 
 	as.logger.Debug("flush virtual stage")
 	controlPlaneConfiguration := make(map[string]*apisix.ApisixConfiguration)
-	virtualStage := NewVirtualStage(utils.GetUUID(), as.operatorURL)
+	virtualStage := NewVirtualStage(version.Commit, as.operatorURL)
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply pkg/apisix/synchronizer/synchronizer.go]: done

File: pkg/apisix/synchronizer/synchronizer.go [hunk 3]
```diff
@@ -45,23 +45,23 @@ type ApisixConfigurationSynchronizer struct {
 	store    ApisixConfigStore
 	flushMux sync.Mutex
 
-	operatorURL string
+	apisixHealthZURI string
```
Comment (wklken) [inline pkg/apisix/synchronizer/synchronizer.go:48]: apisixHealthZURI => apisixHealthzURI

Comment (Han-Ya-Jun) [reply pkg/apisix/synchronizer/synchronizer.go]: done

**Pull request merged**